File: day01/first.py
f = open("data.txt", "r")
lines = f.readlines()

left = []
right = []

for line in lines:
    line = line.strip()
    if line:
        n1, n2 = map(int, line.split(";"))
        left.append(n1)
        right.append(n2)

left.sort()
right.sort()

d = 0
for l, r in zip(left, right):
    d += abs(l - r)

print(d)

File: day02/first.lua
local file = io.open("data.txt", "r")

if not file then
    os.exit(84)
end

local array = {}
local safe = 0

for line in file:lines() do
    local row = {}
    for num in string.gmatch(line, "%S+") do
        table.insert(row, tonumber(num))
    end
    table.insert(array, row)
end

file:close()

local function is_safe_line(line)
    if #line < 2 then
        return false
    end

    local incr = nil

    for i = 1, #line - 1 do
        local diff = line[i + 1] - line[i]

        if math.abs(diff) > 3 then
            return false
        end

        if incr == nil then
            if diff > 0 then
                incr = true
            elseif diff < 0 then
                incr = false
            end
        end
        if (incr and diff <= 0) or (not incr and diff >= 0) then
            return false
        end
    end
    return true
end

for i = 1, #array do
    if is_safe_line(array[i]) then
        safe = safe + 1
    end
end

print(safe)
